db_test.go

```diff
@@ -1301,3 +1301,88 @@ func TestInitializeHeadTimestamp(t *testing.T) {
 		testutil.Equals(t, int64(15000), db.head.MaxTime())
 	})
 }
+
+func TestDB_LabelNames(t *testing.T) {
+	tests := []struct {
+		// We are using 2 sets of input and expected results in order to test the following phases:
+		//     1. Samples only in head.
+		//     2. Samples only on disk.
+		//     3. Samples both in head and disk. (2 sets is helpful in this)
+		sampleLabels1 [][2]string
+		sampleLabels2 [][2]string
+		exp1          []string // after adding sampleLabels1.
+		exp2          []string // after adding sampleLabels1 and sampleLabels2.
+	}{
+		{
+			sampleLabels1: [][2]string{
+				[2]string{"name1", ""},
+				[2]string{"name3", ""},
+				[2]string{"name2", ""},
+			},
+			sampleLabels2: [][2]string{
+				[2]string{"name4", ""},
+				[2]string{"name1", ""},
+			},
+			exp1: []string{"name1", "name2", "name3"},
+			exp2: []string{"name1", "name2", "name3", "name4"},
+		},
+		{
+			sampleLabels1: [][2]string{
+				[2]string{"name2", ""},
+				[2]string{"name1", ""},
+				[2]string{"name2", ""},
+			},
+			sampleLabels2: [][2]string{
+				[2]string{"name6", ""},
+				[2]string{"name0", ""},
+			},
+			exp1: []string{"name1", "name2"},
+			exp2: []string{"name0", "name1", "name2", "name6"},
+		},
+	}
+
+	blockRange := DefaultOptions.BlockRanges[0]
+	// Appends samples into the database.
+	appendSamples := func(db *DB, mint, maxt int64, sampleLabels [][2]string) {
+		app := db.Appender()
+		for i := mint; i <= maxt; i++ {
+			for _, tuple := range sampleLabels {
+				label := labels.FromStrings(tuple[0], tuple[1])
+				_, err := app.Add(label, i*blockRange, 0)
+				testutil.Ok(t, err)
+			}
+		}
+		err := app.Commit()
+		testutil.Ok(t, err)
+	}
+	// Tests for expected label names.
+	testEquals := func(db *DB, exp []string) {
```

Comment:
should be marked as a helper func so if it fails it shows the relevant test line number
https://www.tobstarr.com/2017/06/16/better-test-helpers-in-go/

Reply:
Didn't know about that, thanks! I added.

Reply:
yeah @simonpasquier did this in one PR I reviewed.